src/huggingface_hub/utils/_http.py

```diff
@@ -29,6 +32,139 @@
 logger = logging.get_logger(__name__)
 
 
+@dataclass
+class SessionParameters:
```

Comment:
I am not too sure whether this is better than the alternative of just leaving the kwargs up to the user and for them to check the docs of request, but I figure the autocomplete + dataclass can prevent typos from happening and help create objects simpler.

Reply:
Yes as you guessed, I did it like this to help with the autocomplete and let users know what the possible parameters are. But I can remove it if you think it adds complexity for nothing.

Note that the user never sees this "SessionParameters" object since params are passed to `configure_http_backend` one by one. This dataclass is only used as the global variable to configure sessions (I could use a simple dictionary)